File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 返回全部列表
     */
    List<TbBrand> findAll();

    /**
     * 返回分页列表
     */
    PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    /**
     * 新增品牌
     */
    void add(TbBrand brand);

    /**
     * 根据ID查询
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌
     */
    void update(TbBrand brand);

    /**
     * 删除品牌
     */
    void delete(Long[] ids);

    /**
     * 品牌下拉框数据
     */
    List<Map> selectOptionList();
}

File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };
    this.search = function (page, rows, searchEntity) {
        return $http.post("../brand/search.do?pageNum=" + page + "&pageSize=" + rows, searchEntity);
    };
    this.add = function (entity) {
        return $http.post("../brand/add", entity);
    };
    this.update = function (entity) {
        return $http.post("../brand/update", entity);
    };
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };
    this.delete = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);
    };
    this.selectOptionList = function () {
        return $http.get("/brand/selectOptionList.do");
    }
});